Ch/localizing files

GlowingEgg opened the pull request:
Makes the pictures and videos persist across activities and sessions.

GlowingEgg committed:
Converts Uris returned from gallery to files so they're always accessible.
Changes the Moment to store file paths, instead of uri strings, for the videos and pictures and modifies code that deals with those fields to account for files.

GlowingEgg committed:
Converts Uris returned from gallery to files so they're always accessible.
Changes the Moment to store file paths, instead of uri strings, for the videos and pictures and modifies code that deals with those fields to account for files.

GlowingEgg committed:
Fixes the film your own video bit to act the same as retrieving from gallery.
Also stops having the FileDealer accept a context in the constructor.

GlowingEgg committed:
Fixes the film your own video bit to act the same as retrieving from gallery.
Also stops having the FileDealer accept a context in the constructor.

0xc00010ff commented on app/src/main/java/com/tikkunolam/momentsintime/FileDealer.java:
```diff
@@ -9,25 +9,30 @@
 
 
-    public static String getPath(final Context context, final Uri uri) {
+    public String getPath(final Context context, final Uri uri) {
```
👍 Glad we're getting over singletons and static methods.

0xc00010ff commented on app/src/main/java/com/tikkunolam/momentsintime/InterviewingActivity.java:
```diff
@@ -95,6 +105,20 @@ protected void onCreate(Bundle savedInstanceState) {
+
+                    // Should we show an explanation?
+                    if (shouldShowRequestPermissionRationale(
+                            Manifest.permission.READ_EXTERNAL_STORAGE)) {
```
Just nit picking, but this is pretty broken style (single argument on its own line)

0xc00010ff commented on app/src/main/java/com/tikkunolam/momentsintime/MainActivity.java:
```diff
@@ -61,27 +81,314 @@ private void setupTabLayout() {
 
+    private class AsyncDeleteMoment extends AsyncTask<Moment, Void, Boolean> {
+
+        protected Boolean doInBackground(Moment... moment) {
```
Kind of odd that we're taking variadic args, but still only grabbing moment[0]

0xc00010ff commented on app/src/main/java/com/tikkunolam/momentsintime/MakeAMomentActivity.java:
```diff
@@ -567,6 +618,20 @@ public void onDescriptionPromptClick() {
+            // Should we show an explanation?
+            if (shouldShowRequestPermissionRationale(
+                    Manifest.permission.READ_EXTERNAL_STORAGE)) {
+                // Explain to the user why we need to read the contacts
```
🤔

0xc00010ff commented on app/src/main/java/com/tikkunolam/momentsintime/MyMomentsFragment.java:
```diff
@@ -296,4 +295,60 @@ public void onMessageEvent(UploadFinishedMessage event) {
+        SharedPreferences sharedPreferences = getContext().getApplicationContext().getSharedPreferences(mSharedPreferencesName, MODE_PRIVATE);
+
+        Boolean shouldSwitch = sharedPreferences.getBoolean(mHasFailedFlagName, false);
+
```
‼️ Where do we reset this? For safety you should probably reset it immediately after grabbing it.

0xc00010ff commented on app/src/main/java/com/tikkunolam/momentsintime/StateMomentCardHolder.java:
```diff
@@ -39,20 +62,300 @@ public StateMomentCardHolder(Context context, View view) {
+        protected Moment doInBackground(Moment... momentArg) {
+
+            // take the Moment
+            Moment moment = momentArg[0];
```
Variadic single-grab thing again. Not a big deal but philosophically misaligned

GlowingEgg committed:
Fixes some issues and clarifies some things.
Puts the shouldShowRequestPermissionRationale arguments on one line for clarity. Adds some comments to hopefully justify the heavy-handed variadic argument use in AsyncTasks to future developers. Adds a line to flip back the flag that notifies if an upload failed, after checking it.
